From: Ralf Baechle <ralf@linux-mips.org>
Date: Fri, 12 Jun 2009 16:28:00 +0000 (+0100)
Subject: MIPS: Fix typo resulting in far too long ndelay times.
X-Git-Tag: linux-2.6.30~1
X-Git-Url: http://www.linux-mips.org/git?p=linux.git;a=commitdiff_plain;h=a7258465ba52a3e3bb38e8d30b5425fbd3e16386

MIPS: Fix typo resulting in far too long ndelay times.

Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
---

diff --git a/arch/mips/include/asm/delay.h b/arch/mips/include/asm/delay.h
index a07e51b..d2d8949 100644
--- a/arch/mips/include/asm/delay.h
+++ b/arch/mips/include/asm/delay.h
@@ -15,7 +15,7 @@ extern void __delay(unsigned int loops);
 extern void __ndelay(unsigned int ns);
 extern void __udelay(unsigned int us);
 
-#define ndelay(ns) __udelay(ns)
+#define ndelay(ns) __ndelay(ns)
 #define udelay(us) __udelay(us)
 
 /* make sure "usecs *= ..." in udelay do not overflow. */
From: Atsushi Nemoto <nemoto@toshiba-tops.co.jp>
Date: Tue, 9 Jun 2009 02:12:48 +0000 (+0900)
Subject: MIPS: Fix __ndelay build error and add 'ull' suffix for 32-bit kernel
X-Git-Tag: linux-2.6.30~2
X-Git-Url: http://www.linux-mips.org/git?p=linux.git;a=commitdiff_plain;h=60bd2aaf47ba20fd99901b8bcd2bebffe37b166d

MIPS: Fix __ndelay build error and add 'ull' suffix for 32-bit kernel

Signed-off-by: Atsushi Nemoto <anemo@mba.ocn.ne.jp>
Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
---

diff --git a/arch/mips/lib/delay.c b/arch/mips/lib/delay.c
index f69c6b5..6b3b1de 100644
--- a/arch/mips/lib/delay.c
+++ b/arch/mips/lib/delay.c
@@ -43,7 +43,7 @@ void __udelay(unsigned long us)
 {
 	unsigned int lpj = current_cpu_data.udelay_val;
 
-	__delay((us * 0x000010c7 * HZ * lpj) >> 32);
+	__delay((us * 0x000010c7ull * HZ * lpj) >> 32);
 }
 EXPORT_SYMBOL(__udelay);
 
@@ -51,6 +51,6 @@ void __ndelay(unsigned long ns)
 {
 	unsigned int lpj = current_cpu_data.udelay_val;
 
-	__delay((us * 0x00000005 * HZ * lpj) >> 32);
+	__delay((ns * 0x00000005ull * HZ * lpj) >> 32);
 }
 EXPORT_SYMBOL(__ndelay);
